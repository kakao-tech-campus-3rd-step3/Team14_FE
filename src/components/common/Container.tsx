interface ContainerProps {
  children: React.ReactNode;
}

/**
 * Container 컴포넌트
 * @param children - 컴테이너 내부에 표시될 콘텐츠
 */
const Container = ({ children }: ContainerProps) => {
  return (
    <div className="w-full h-full flex flex-col items-center bg-gray-100">
      <div className="min-h-screen pt-12 pb-15 max-w-[480px] w-full relative bg-white ">
        {children}
      </div>
    </div>
  );
};

export default Container;
