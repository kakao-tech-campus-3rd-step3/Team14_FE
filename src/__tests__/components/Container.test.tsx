import { render, screen } from '@testing-library/react';
import Container from '@/components/common/Container';

describe('Container 컴포넌트', () => {
  describe('스냅샷 테스트', () => {
    test('Container가 렌더링된다', () => {
      // Given: Container 컴포넌트가 주어졌을 때
      // When: Container 컴포넌트를 렌더링하면
      // Then: Container가 렌더링된다
      const { container } = render(<Container>테스트</Container>);
      expect(container).toMatchSnapshot();
    });
  });
  describe('children 테스트', () => {
    test('children이 올바르게 렌더링된다', () => {
      // Given: children 요소가 주어졌을 때
      const childText = '테스트 컨텐츠';

      // When: children을 포함한 Container 컴포넌트를 렌더링하면
      render(
        <Container>
          <div>{childText}</div>
        </Container>,
      );

      // Then: children 요소가 화면에 표시돼야 한다
      expect(screen.getByText(childText)).toBeVisible();
    });
  });

  describe('기능 테스트', () => {
    test('여러 children을 올바르게 렌더링한다', () => {
      // Given: 여러 children 요소가 주어졌을 때
      // When: 여러 children으로 Container 컴포넌트를 렌더링하면
      render(
        <Container>
          <div>첫 번째 요소</div>
          <div>두 번째 요소</div>
          <div>세 번째 요소</div>
        </Container>,
      );

      // Then: 모든 children이 화면에 표시돼야 한다
      expect(screen.getByText('첫 번째 요소')).toBeVisible();
      expect(screen.getByText('두 번째 요소')).toBeVisible();
      expect(screen.getByText('세 번째 요소')).toBeVisible();
    });
  });
});
