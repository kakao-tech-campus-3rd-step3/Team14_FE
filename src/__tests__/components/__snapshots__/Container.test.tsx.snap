// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container 컴포넌트 > 스냅샷 테스트 > Container가 렌더링된다 1`] = `
<div>
  <div
    class="w-full h-full flex flex-col items-center bg-gray-100"
  >
    <div
      class="min-h-screen pt-12 pb-15 max-w-[480px] w-full relative bg-white "
    >
      테스트
    </div>
  </div>
</div>
`;
